java-manta-client/src/main/java/com/joyent/manta/client/crypto/AesCipherDetailsFactory.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2017, Joyent, Inc. All rights reserved.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.joyent.manta.client.crypto;
+
+import javax.crypto.Cipher;
+import java.security.NoSuchAlgorithmException;
+
+/**
+ * Class for constructing AesCipherDetails which may or may not be useable in the current runtime.
+ *
+ * @author <a href="https://github.com/tjcelaya">Tomas Celayac</a>
+ * @since 3.0.0
+ */
+public final class AesCipherDetailsFactory {
+
+    /**
+     * Private constructor.
+     */
+    private AesCipherDetailsFactory() {
+    }
+
+    /**
+     * Supported AES Cipher modes.
+     */
+    @SuppressWarnings("checkstyle:JavadocVariable")
+    public enum CipherMode {
+        CBC,
+        CTR,
+        GCM
+    }
+
+    /**
+     * Default maximum key length.
```

Comment:
We expect the "normal" thing to be using the unlimited encryption thought, right? I'm not sure "default" is the best term.